packages/calcite-components/src/components/combobox-item/combobox-item.tsx

```diff
@@ -252,12 +259,27 @@ export class ComboboxItem implements ConditionalSlotComponent, InteractiveCompon
             <li class={classes} id={this.guid} onClick={this.itemClickHandler}>
               {this.renderSelectIndicator(showDot, iconPath)}
               {this.renderIcon(iconPath)}
-              <span class="title">{this.textLabel}</span>
+              <span class="title">{this.renderTextContent()}</span>
             </li>
             {this.renderChildren()}
           </div>
         </InteractiveContainer>
       </Host>
     );
   }
+
+  private renderTextContent(): string | (string | VNode)[] {
+    if (!this.filterTextMatchPattern) {
+      return this.textLabel;
+    }
+
+    const parts: (string | VNode)[] = this.textLabel.split(this.filterTextMatchPattern);
+
+    if (parts.length > 1) {
+      // we only highlight the first match
+      parts[1] = <span class={CSS.filterMatch}>{parts[1]}</span>;
+    }
+
+    return parts;
+  }
```

Comment:
not sure if this is helpful, but this is what we have for the maps sdk.

```tsx
const regexContainsHTML = /<[a-z/][\s\S]*>/i;

const boldKeywords = (text: string, keywords: string): string => {
  const escapedKeywords = escapeRegExpString(keywords);
  const splitKeywords = escapedKeywords.split(/\s/);
  const pattern = new RegExp(`(${splitKeywords.join("|")})`, "gi");
  return text.replace(pattern, (match) => `<strong>${match}</strong>`);
};
```

... later


```
return regexContainsHTML.test(text) ? text : boldKeywords(text, searchTerm);
```

Reply:
We may want to follow the same logic since Search widget would likely want the same bold experience.

Reply:
Great suggestion. Will update!

Reply:
Actually, could we discuss this in a follow-up issue? There are some differences in implementation and UI/UX that I think need to be considered first, such as:

* combobox filtering is immediate, search has minimum chars before showing matches
* combobox highlights the first occurrence, search highlights all matches
* combobox items do not support HTML as content, search does

There could be others, but these are the ones I noticed.

Reply:
I did add the regexp char escape. 🏃🔠 Thanks for the bringing that up!

Reply:
> combobox filtering is immediate, search has minimum chars before showing matches

I think this could be handled when we support combobox with dynamic data

> combobox highlights the first occurrence, search highlights all matches

Shouldn't combobox highlight all matches? Why does it only highlight the first occurrence?

> combobox items do not support HTML as content, search does

Yes, we can ignore this one. It doesn't do regex if there is HTML involved.

Reply:
> Shouldn't combobox highlight all matches? Why does it only highlight the first occurrence?

This closely matches the current filtering behavior. It returns on the first match. I also did this to minimize elements updated as the user types since we don't have virtualization alternatives and showing all matches < 3 chars seems unnecessary. 

FWIW, I'm fine revisiting ☝️, but I would like design to give a pass before doing so. It would be an incremental enhancement this way vs rolling it back.

Reply:
Yeah sounds good.